proposals/dao/v2Phase1-part1.ts

```diff
@@ -0,0 +1,144 @@
+import {
+  BondingCurve,
+  CollateralizationOracleKeeper,
+  Core,
+  ERC20CompoundPCVDeposit,
+  EthBondingCurve,
+  PCVEquityMinter,
+  RatioPCVController,
+  TribeReserveStabilizer,
+  UniswapPCVDeposit
+} from '@custom-types/contracts';
+import { RunUpgradeFunc, SetupUpgradeFunc, TeardownUpgradeFunc, ValidateUpgradeFunc } from '@custom-types/types';
+import '@nomiclabs/hardhat-ethers';
+import chai from 'chai';
+import CBN from 'chai-bn';
+import hre, { ethers } from 'hardhat';
+
+before(() => {
+  chai.use(CBN(ethers.BigNumber));
+});
+
+/*
+
+V2 Phase 1 Upgrade
+
+Part 1 - Deploys the PCV deposits we have to swap out, the new ETH bonding curve, and the ratio PCV controller.
+         Grants minter roles to the pcv deposits & the bonding curve, and pcv controller role to the ratio pcv controller.
+         Sets bonding curve minting cap maximum for eth bonding curve, and updates the dpi bonding curve allocation. Finally,
+         moves pcv from the old eth & dpi uni pcv deposits into the new ones.
+
+----- PART 1 -----
+
+DEPLOY ACTIONS:
+1. ETH Uni PCV Deposit
+2. DPI Uni PCV Deposit
+3. ETH Bonding Curve
+4. Ratio PCV Controller
+
+DAO ACTIONS:
+1. Grant Minter role to new ETH BondingCurve
+2. Grant Minter role to new ETH Uni PCV Deposit
+3. Grant Minter role to new DPI Uni PCV Deposit
+4. Grant PCV Controller role to new RatioPCVController
+5. Set ETH Bonding Curve Minting Cap Max
+6. Update DPI Bonding Curve allocation
+7. Move PCV from old ETH Uni PCV Deposit to new
+8. Move PCV from old DPI Uni PCV Deposit to new
+9. Sets allocation on the DPI bonding curve.
+10. Sets mint cap on the new eth bonding curve.
+11. Revokes the minter role from the old uniswap pcv deposit.
+12. Revokes the minter role from the old dpi uniswap pcv dpeosit.
+13. Revokes the minter role from the old eth bonding curve.
+14. Calls deposit() on the dpi uniswap pcv deposit.
+15. Calls deposit() on the old uniswap pcv deposit.
+16. Revokes the pcv controller role from the old ratio pcv controller.
+17. Revokes the pcv controller role from the old uniswap pcv controller.
+18. Revokes the minter role from the old uniswap pcv controller.
+
+*/
+
+export const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  const { timelock } = addresses;
+
+  await hre.network.provider.request({
+    method: 'hardhat_impersonateAccount',
+    params: [timelock]
+  });
+};
+
+export const run: RunUpgradeFunc = async (addresses, oldContracts, contracts, logging = false) => {
```

Comment:
nit, its helpful to include the numbering in comments in the run section for tracking